common/src/stack/switch/command/list/switch/partition/__init__.py

```diff
@@ -75,6 +81,11 @@ def run(self, params, args):
 		sw_select += ' ORDER BY nodes.name'
 
 		self.beginOutput()
-		for line in self.db.select(sw_select, vals):
-			self.addOutput(line[0], (line[1], '0x{0:04x}'.format(line[2]), line[3]))
+		if self.str2bool(expanded):
+			for switch_name in switches:
+				model = self.getHostAttr(switch_name, 'component.model')
+				self.runImplementation(model, [switch_name])
```

Comment:
Consider refactoring to not have the implementation dictate the output format, or whether the results are output at all.